flytekit/annotated/type_engine.py

```diff
@@ -310,7 +310,7 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
             for k, v in lv.map.literals.items():
                 py_map[k] = TypeEngine.to_python_value(ctx, v, tp[1])
             return py_map
-        if lv and lv.scalar and lv.scalar.generic:
+        if lv is not None and lv.scalar is not None and lv.scalar.generic is not None:
```

Comment:
If this statement needs to be `if lv is not None` instead of just `if lv`, then can you add a comment?   Don't want someone refactoring this to the latter in the future.